templates/components/engines/python/agent/tools/query_engine.py

```diff
@@ -1,28 +1,106 @@
 import os
-from typing import Optional
+from typing import Any, List, Optional, Sequence
 
+from llama_index.core.base.base_query_engine import BaseQueryEngine
+from llama_index.core.base.response.schema import RESPONSE_TYPE, Response
+from llama_index.core.multi_modal_llms import MultiModalLLM
+from llama_index.core.prompts.base import BasePromptTemplate
+from llama_index.core.prompts.default_prompt_selectors import (
+    DEFAULT_TREE_SUMMARIZE_PROMPT_SEL,
+)
+from llama_index.core.query_engine import (
+    RetrieverQueryEngine,
+)
+from llama_index.core.query_engine.multi_modal import _get_image_and_text_nodes
+from llama_index.core.response_synthesizers import TreeSummarize
+from llama_index.core.response_synthesizers.base import QueryTextType
+from llama_index.core.schema import (
+    ImageNode,
+    NodeWithScore,
+)
 from llama_index.core.tools.query_engine import QueryEngineTool
 
+from app.settings import get_multi_modal_llm
 
-def create_query_engine(index, **kwargs):
+
+class MultiModalSynthesizer(TreeSummarize):
```

Comment:
use base synthesizer and pass a responsesynthesizer to the constructor. then you can delegate to this synthesizer